File: Backend/SOS-Clean-House-back/src/main/java/com/SOS_Clean_House_back/model/Agendamento.java
package com.SOS_Clean_House_back.model;

import com.fasterxml.jackson.annotation.*;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;
import java.util.Date;
import java.util.List;

@Entity
@AllArgsConstructor @NoArgsConstructor @Data @Builder
public class Agendamento {

    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private Integer id;
    private Boolean status;
    private String horarioAgendamento;
    private String horarioAgendamentoFim;
    private String valorAgendamento;
    private List<String> atividadesAgendadas;

    @JsonFormat(shape = JsonFormat.Shape.STRING, pattern = "yyyy-MM-dd'T'HH:mm:ss.SSSX", timezone = "America/Sao_Paulo")
    private Date dataAgendamento;

    @JsonFormat(shape = JsonFormat.Shape.STRING, pattern = "yyyy-MM-dd'T'HH:mm:ss.SSSX", timezone = "America/Sao_Paulo")
    private Date dataServico;

    @JsonProperty(access = JsonProperty.Access.READ_ONLY)
    @OneToOne(mappedBy = "enderecoAgendamento")
    private EnderecoServico enderecoServico;

    @ManyToOne
    @JoinColumn(name = "servico_id")
    private Servico servico;

    @ManyToOne
    @JoinColumn(name = "cliente_id")
    private Cliente cliente;

}

File: Frontend/SOS-Clean-House-front/src/app/componentes/interface/Agendamento.ts
import { Cliente } from "./Cliente";
import { DiaCalendario } from "./DiaCalendario";
import { EnderecoServico } from "./EnderecoServico";
import { Prestador } from "./Prestador";
import { Servico } from "./Servico";

export interface Agendamento {
    id: number;
    status: boolean;
    dataAgendamento: Date;
    horarioAgendamento: string;
    horarioAgendamentoFim: string;
    dataServico: Date;
    enderecoServico: EnderecoServico;
    servico: Servico;
    cliente: Cliente;
    prestador: Prestador;
    atividadesAgendadas: string[];
    valorAgendamento: string;
  }